samples/python/app.py

```diff
@@ -0,0 +1,94 @@
+"""
+This example integrates the IdPyOIDC library with the MIRACL Trust platform.
+"""
+import os
+import random
+from urllib.parse import parse_qs
+
+from flask import Flask, redirect, request
+from idpyoidc.client.rp_handler import RPHandler
+
+session_storage = {}
+proxy_host = os.environ.get("PROXY_HOST")
+proxy_port = os.environ.get("PROXY_PORT")
+
+
+def new_app():
+    """
+    Creates an instance of a flask application, relying party handler and a relying party client.
+    """
+    application = Flask(__name__)
+    client_configs = {
+        "miracl": {
+            "issuer": "https://api.mpin.io/",
+            "client_id": os.environ.get("CLIENT_ID"),
+            "client_secret": os.environ.get("CLIENT_SECRET"),
+            "client_type": "oidc",
+            "redirect_uris": ["http://localhost:8000/login"],
+        }
+    }
+    rp_handler = RPHandler(
+        client_configs=client_configs,
+        httpc_params=(
+            {
+                "proxies": {
+                    "http": f"{proxy_host}:{proxy_port}",
+                    "https": f"{proxy_host}:{proxy_port}",
+                }
+            }
+            if proxy_host is not None and proxy_port is not None
+            else None
```

Comment:
What do those lines do?

Reply:
They set httpc_params to None if proxy_host or proxy_port  is None, if they are both not none, then the dictionary from lines 33 to 38 is used as a value for httpc_params.